File: Binary_Search.c
#include<stdio.h>
int main()
{
   int n=5;
    int arr[n];
    printf("Enter 5 elements\n");
    for(int i=0;i<n;i++){
        scanf("%d",&arr[i]);
    }
    int first,last,middle,search;
    printf("Enter search element\n");
    scanf("%d",&search);
    first=0;
    last=n-1;
    middle=(first+last)/2;
    while(first<=last)
    {
        if(arr[middle]<search)
        {
            first=middle+1;
        }
        else if(search==arr[middle])
        {
            printf("%d Found at index %d",search,middle+1);
            break;
        }
        else{
            last=middle-1;
        }
        middle=(first+last)/2;
    }
}

File: Binary_Search.java

public class Binary_Search {
public static void main(String args[])
{
	int n=8;
	int[] a= {15,25,33,49,52,64,77,9};
	int high=n;
	int low=0;
	int key=49;
	int mid =0;
	while(low<=high)
	{
		mid = (high+low)/2;
		if(a[mid]== key)
		{
			System.out.print("Found at index "+(mid+1));
			break;
		}
		else if(key>a[mid])
		{
			low=mid+1;
		}
		else
		{
			high=mid-1;
		}
	}
}
}
